core/src/commonMain/kotlin/com/sourcepoint/mobile_core/Coordinator.kt

```diff
@@ -38,4 +59,244 @@ class Coordinator(
         repository.cachedMetaData = metaDataResponse.toString()
         return message
     }
+
+    private fun handleGetChoiceAll(response: ChoiceAllResponse, campaign: SPCampaignType) {
+        if (response.gdpr != null && campaign == SPCampaignType.Gdpr) {
+            state.gdpr?.dateCreated = response.gdpr.dateCreated
+            state.gdpr?.expirationDate = response.gdpr.expirationDate
+            state.gdpr?.tcData = response.gdpr.tcData?: emptyMap()
+            state.gdpr?.grants = response.gdpr.grants
+            state.gdpr?.euconsent = response.gdpr.euconsent
+            state.gdpr?.consentStatus = response.gdpr.consentStatus
+            state.gdpr?.childPmId = response.gdpr.childPmId
+            state.gdpr?.gcmStatus = response.gdpr.gcmStatus
+        }
+        if (response.ccpa != null && campaign == SPCampaignType.Ccpa) {
+            state.ccpa?.dateCreated = response.ccpa.dateCreated
+            state.ccpa?.expirationDate = response.ccpa.expirationDate
+            state.ccpa?.status = response.ccpa.status
+            state.ccpa?.gppData = response.ccpa.gppData
+            state.ccpa?.uspstring = response.ccpa.uspstring
+        }
+        if (response.usnat != null && campaign == SPCampaignType.UsNat) {
+            state.usNat?.dateCreated = response.usnat.dateCreated
+            state.usNat?.expirationDate = response.usnat.expirationDate
+            state.usNat?.consentStatus = response.usnat.consentStatus
+            state.usNat?.gppData = response.usnat.gppData
+            state.usNat?.consentStrings = response.usnat.consentStrings
+            state.usNat?.consentStatus?.consentedToAll = response.usnat.consentedToAll
+            state.usNat?.consentStatus?.rejectedAny = response.usnat.rejectedAny
+            state.usNat?.consentStatus?.granularStatus?.gpcStatus = response.usnat.gpcEnabled
+        }
```

Comment:
Check my comment below regarding `copy`. Here, instead of modifying each attribute, we can use the `copy` function to reassign the entire object.